safe/gui/tools/multi_exposure_dialog.py

```diff
@@ -0,0 +1,243 @@
+# coding=utf-8
+
+"""Multi Exposure Tool."""
+
+import logging
+
+
+from PyQt4.QtGui import (
+    QDialog, QComboBox, QLabel, QDialogButtonBox, QApplication, QSizePolicy)
+
+from qgis.core import QgsMapLayerRegistry
+from qgis.utils import iface
+
+from safe import messaging as m
+from safe.common.exceptions import (
+    NoKeywordsFoundError,
+    KeywordNotFoundError,
+    MetadataReadError,
+)
+from safe.common.signals import send_error_message
+from safe.definitions.constants import (
+    inasafe_keyword_version_key,
+    ANALYSIS_FAILED_BAD_INPUT,
+    entire_area_item_aggregation,
+)
+from safe.definitions.exposure import exposure_all
+from safe.definitions.layer_purposes import (
+    layer_purpose_hazard,
+    layer_purpose_exposure,
+    layer_purpose_aggregation,
+)
+from safe.gui.analysis_utilities import add_impact_layers_to_canvas
+from safe.gui.gui_utilities import layer_from_combo, add_ordered_combo_item
+from safe.gui.widgets.message import enable_messaging, send_static_message
+from safe.impact_function.multi_exposure_wrapper import (
+    MultiExposureImpactFunction)
+from safe.messaging import styles
+from safe.utilities.gis import qgis_version
+from safe.utilities.keyword_io import KeywordIO
+from safe.utilities.qt import disable_busy_cursor, enable_busy_cursor
+from safe.utilities.resources import (
+    get_ui_class,
+)
+from safe.utilities.settings import setting
+from safe.utilities.utilities import (
+    is_keyword_version_supported,
+)
+from safe.utilities.i18n import tr
+
+LOGGER = logging.getLogger('InaSAFE')
+FORM_CLASS = get_ui_class('multi_exposure_dialog_base.ui')
+
+INFO_STYLE = styles.BLUE_LEVEL_4_STYLE
+LOGO_ELEMENT = m.Brand()
+
+
+class MultiExposureDialog(QDialog, FORM_CLASS):
+
+    """Dialog for multi exposure tool."""
+
+    def __init__(self, parent=None):
+        """Constructor for the multi exposure dialog.
+
+        :param parent: Parent widget of this dialog.
+        :type parent: QWidget
+        """
+        QDialog.__init__(self, parent)
+        self.parent = parent
+        self.setupUi(self)
+        self.combos_exposures = {}
+        self.keyword_io = KeywordIO()
+        self._create_exposure_combos()
+        self._multi_exposure_if = None
+
+        enable_messaging(self.message_viewer, self)
+
+        # Fix for issue 1699 - cancel button does nothing
+        cancel_button = self.button_box.button(QDialogButtonBox.Cancel)
```

Comment:
__[flake8]__
* [F841] local variable 'cancel_button' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->